elastic/kibana
# [Lens] Make Lens intervals default value adapt to histogram:maxBars Advanced Setting changes

## Summary

Fixes #83420

This PR uses the UISettings `histogram:maxBars` value only as fallback in case the user hasn't changed the range input yet.
This makes the "default value" for the Interval UI dynamically change if a user changes the `histogram:maxBars` value in the Kibana *Advanced Settings*.

### To test the feature

* Create a chart using the Interval operation
	* Drop a numeric field in the Horizontal axis area (preferably drop something with bigger ranges like `price` or `bytes`)
	* Open and close the Interval panel (do not change the range value).
	* Drop the `Records` field in the Vertical axis area
* Write down or take a screenshot of the chart to remember the interval size 
* Save the chart as `myIntervalChart1`
* Create a secondary chart, this time like the previous, but **do not open the Horizontal axis dimension panel**.
* Write down or take a screenshot of the chart to remember the interval size also for this one
* Save it as `myIntervalChart2`
* Go to `Stack Management > Advanced Settings`
* Search for `maxBars` and change the value from 100 to 50
* Open the `myIntervalChart1` and check that the interval has now changed (compare with the note/screenshot from before)
* Open the `myIntervalChart2` and check that the interval has now changed (compare with the note/screenshot from before). It should be the same size as `myIntervalChart1` one.

Tests have been re-adapted to follow the new behaviour.

### Checklist
